src/main/java/com/exasol/adapter/dynamodb/documentfetcher/dynamodb/DynamodbQueryDocumentFetcher.java

```diff
@@ -3,32 +3,168 @@
 import java.util.Map;
 import java.util.stream.Stream;
 
-import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
-import com.amazonaws.services.dynamodbv2.model.AttributeValue;
-import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.exasol.adapter.dynamodb.documentnode.DocumentValue;
+import com.exasol.adapter.dynamodb.documentnode.dynamodb.DynamodbNodeVisitor;
+
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.QueryRequest;
 
 /**
  * This class represents a DynamoDB {@code QUERY} operation.
  */
 public class DynamodbQueryDocumentFetcher extends AbstractDynamodbDocumentFetcher {
-    private static final long serialVersionUID = -6805336709119451973L;
-    private final QueryRequest queryRequest;
+    private static final long serialVersionUID = -810868396675247321L;
+
+    private final GenericTableAccessParameters genericParameters;
+    private final String indexName;
+    private final String keyConditionExpression;
 
     /**
      * Create an a {@link DynamodbQueryDocumentFetcher}.
      *
-     * @param queryRequest DynamoDB request
+     * @param genericParameters      generic parameters (not query specific)
+     * @param indexName              name of the index to scan
+     * @param keyConditionExpression condition expression for key columns
      */
-    DynamodbQueryDocumentFetcher(final QueryRequest queryRequest) {
-        this.queryRequest = queryRequest;
+    private DynamodbQueryDocumentFetcher(final GenericTableAccessParameters genericParameters, final String indexName,
+            final String keyConditionExpression) {
+        this.genericParameters = genericParameters;
+        this.indexName = indexName;
+        this.keyConditionExpression = keyConditionExpression;
+    }
+
+    static Builder builder() {
+        return new Builder();
     }
 
     QueryRequest getQueryRequest() {
-        return this.queryRequest;
+        final QueryRequest.Builder builder = QueryRequest.builder().tableName(this.genericParameters.getTableName());
+        if (this.genericParameters.hasExpressionAttributeNames()) {
+            builder.expressionAttributeNames(this.genericParameters.getExpressionAttributeNames());
+        }
+        if (this.genericParameters.hasFilterExpression()) {
+            builder.filterExpression(this.genericParameters.getFilterExpression());
+        }
+        if (this.genericParameters.hasExpressionAttributeValues()) {
+            builder.expressionAttributeValues(this.genericParameters.getExpressionAttributeValues());
+        }
+        if (this.genericParameters.hasProjectionExpression()) {
+            builder.projectionExpression(this.genericParameters.getProjectionExpression());
+        }
+        if (this.indexName != null) {
```

Comment:
I just thought: do you really need to check each field here? You have validation in the builder, so the values should also present don't they?

Reply:
Id need it here. In the builder I check if the fields were set. (except indeName since its the only one that is not required)

Here I check if the fields are empty even though they are set. This is necessary since DynamoDB throws an Exception if an empty map was added.